pymc/tests/test_sampling.py

```diff
@@ -865,6 +866,27 @@ def test_idata_extension(self):
         fails = check_multiple_attrs(test_dict, idata)
         assert not fails
 
+    def test_deterministics_out_of_idata(self):
```

Comment:
Does this mean that deterministics now work in inference_data but not multitrace?

Reply:
Ah the idata only contains `a`

Reply:
Can we parametrize this test for both idata/multitrace in any case?

Reply:
They worked before with MultiTrace, but didn't work with InferenceData. This PR fixed that

Reply:
I know that, but it seems useful to test both cases to prevent a future regression when using MultiTrace. Seems like this bug could have happened equally easily the other way around.